daffodil-lib/src/test/scala/org/apache/daffodil/validation/ValidatorSPISupport.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.validation
+
+import java.io.InputStream
+
+import com.typesafe.config.Config
+import org.apache.daffodil.api.ValidationFailure
+import org.apache.daffodil.api.ValidationResult
+import org.apache.daffodil.api.ValidationWarning
+import org.apache.daffodil.api.Validator
+import org.apache.daffodil.api.ValidatorFactory
+
+class PassingValidatorFactory extends ValidatorFactory {
+  def name(): String = PassingValidator.name
+  def make(config: Config): Validator = new TestingValidatorSPI(Seq.empty, Seq.empty)
+}
+object PassingValidator {
+  val name = "passing-validator"
+}
+
+class FailingValidatorFactory extends ValidatorFactory {
+  def name(): String = FailingValidator.name
+  def make(config: Config): Validator = new TestingValidatorSPI(Seq.empty, Seq(ValFail("boom")))
```

Comment:
I'd swear I'm reading this code the second time now. Didn't we already have this above? Or was that Java code?

Reply:
This was a change from the other PR, and yeah there were two similar sets of tests, one in JAPI and one in SAPI.